pkg/client/client.go

```diff
@@ -221,6 +235,45 @@ func (c *Client) runBuild(ctx context.Context, r interface{}, path, plandir, sdk
 	return c.request(ctx, "POST", path, rd, "Content-Type", contentType)
 }
 
+// getFilteredDirectory filters the directory dir according to the
+// ignored files specified in $dir/.testgroundignore. Returns a new
+// directory and a boolean indicating if such directory is temporary
+// and should be cleaned by the caller.
+func getFilteredDirectory(dir string) (string, bool, error) {
+	ignoreFilePath := filepath.Join(dir, ".testgroundignore")
+	_, err := os.Stat(ignoreFilePath)
+
+	if os.IsNotExist(err) {
+		return dir, false, nil
```

Comment:
Don't we want to create a temp dir either way?

Is it possible that the dir files get modified during the run? If so, then if we always create a temp dir, no matter if `.testgroundignore` is provided or not, the behaviour is consistent and the file modifications are not reflected after the run.

Reply:
We also wouldn't need the temp boolean anymore if we always create a temp dir.

Reply:
Sounds good, I liked that first approach, but dropping it make sense, fixed.